exchanges/gateio/gateio_ws_delivery_futures.go

```diff
@@ -78,13 +78,17 @@ func (g *Gateio) GenerateDeliveryFuturesDefaultSubscriptions() (subscription.Lis
 			futuresBalancesChannel,
 		)
 	}
-	pairs, err := g.GetEnabledPairs(asset.DeliveryFutures)
-	if err != nil {
-		return nil, err
-	}
 
 	var subscriptions subscription.List
 	for i := range channelsToSubscribe {
+		pairs, err := g.GetEnabledPairs(asset.DeliveryFutures)
```

Comment:
I think this needs re-looking at. Its now in a loop with an expensive call and if the pairs aren't enabled, continue and call the function again?

My guess here is that you want to return `nil, nil`. So I think you can just do that outside of this loop

Reply:
what 😆

Reply:
y i do dis?